clients/drcachesim/reader/reader.h

```diff
@@ -110,18 +110,25 @@ class reader_t : public std::iterator<std::input_iterator_tag, memref_t>,
     virtual reader_t &
     operator++();
 
-    // Skips records until "count" instruction records have been passed.
-    // This will skip top-level headers for a thread; it is up to the caller
-    // to first observe those before skipping, if needed.  For interleaved-thread
-    // iteration, top-level headers in other threads will be skipped as well
-    // (but generally speaking these are identical to the initial thread).
-    // TODO i#5538: Add access to these header values from #memtrace_stream_t
-    // and document it here.
-    // TODO i#5538: Skipping from the middle will not always duplicate the
-    // last timestamp,cpu.
+    // Skips records until "count" instruction records have been passed.  If any
+    // timestamp (plus cpuid) is skipped, the most recent skipped timestamp will be
+    // duplicated prior to the target instruction.  Top-level headers will still be
+    // observed.  This generally should call pre_skip_instructions() to observe the
+    // headers, perform any fast skipping, and then should call
+    // skip_instructions_with_timestamp() to properly duplicate the prior timestamp.
     virtual reader_t &
     skip_instructions(uint64_t instruction_count);
 
+    virtual void
+    pre_skip_instructions();
```

Comment:
Why are pre_skip_instructions and skip_instructions_with_timestamp public?

Reply:
Shouldn't be: moved to protected.